x/interchainstaking/keeper/callbacks_test.go

```diff
@@ -1607,5 +1611,171 @@ func (suite *KeeperTestSuite) TestDelegationAccountBalanceCallback() {
+		// get chainA context
+		ctx := suite.chainA.GetContext()
+
+		// get zone chainB context
+		zone, _ := quicksilver.InterchainstakingKeeper.GetZone(ctx, suite.chainB.ChainID)
+		zone.DepositAddress.IncrementBalanceWaitgroup()
+		zone.WithdrawalAddress.IncrementBalanceWaitgroup()
+		quicksilver.InterchainstakingKeeper.SetZone(ctx, &zone)
+
+		// get the tx from fixture
+		txWithProofBz := decodeBase64NoErr(localDepositTxFixture)
+		txRes := icqtypes.GetTxWithProofResponse{}
+		err := quicksilver.InterchainQueryKeeper.IBCKeeper.Codec().Unmarshal(txWithProofBz, &txRes)
+		suite.NoError(err)
+
+		// update payload header to ensure we can validate it.
+		txRes.Header.Header.Time = ctx.BlockTime()
+		// setup ClientConsensusState for checking Header validation
+		// Cheat, and set the client state and consensus state for 07-tendermint-0 to match the incoming header.
+		quicksilver.IBCKeeper.ClientKeeper.SetClientState(ctx, "07-tendermint-0", lightclienttypes.NewClientState("gaiatest-1", lightclienttypes.DefaultTrustLevel, time.Hour, time.Hour, time.Second*50, txRes.Header.TrustedHeight, []*ics23.ProofSpec{}, []string{}, false, false))
+		quicksilver.IBCKeeper.ClientKeeper.SetClientConsensusState(ctx, "07-tendermint-0", txRes.Header.TrustedHeight, txRes.Header.ConsensusState())
+
+		// construct request data using txHash
+		requestData := tx.GetTxRequest{
+			// hash of another tx that different from the one in `txFixture`
+			Hash: "2CC0F0C5106F30F5D26ABE8CB93F1EF0CCCE10754207C38B129D76ED3B7C75B2",
+		}
+		resDataBz, err := quicksilver.AppCodec().Marshal(&requestData)
+		suite.NoError(err)
+
+		// initialize the callback
+		err = keeper.DepositTxCallback(quicksilver.InterchainstakingKeeper, ctx, txWithProofBz, icqtypes.Query{ChainId: suite.chainB.ChainID, Request: resDataBz})
+
+		// fail because of txHash mismatch with hash from tx Proof
+		suite.Error(err)
+	})
+	suite.Run("Deposit transaction failed: nil or wrong bytes", func() {
+		suite.SetupTest()
+		suite.setupTestZones()
+
+		// setup quicksilver test app
+		quicksilver := suite.GetQuicksilverApp(suite.chainA)
+		quicksilver.InterchainstakingKeeper.CallbackHandler().RegisterCallbacks()
+
+		// get chainA context
+		ctx := suite.chainA.GetContext()
+
+		// get zone chainB context
+		zone, _ := quicksilver.InterchainstakingKeeper.GetZone(ctx, suite.chainB.ChainID)
+		zone.DepositAddress.IncrementBalanceWaitgroup()
+		zone.WithdrawalAddress.IncrementBalanceWaitgroup()
+		quicksilver.InterchainstakingKeeper.SetZone(ctx, &zone)
+
+		// construct request data using txHash
+		requestData := tx.GetTxRequest{
+			// hash of tx in `txFixture`
+			Hash: "b1f1852d322328f6b8d8cacd180df2b1cbbd3dd64536c9ecbf1c896a15f6217a",
+		}
+		resDataBz, err := quicksilver.AppCodec().Marshal(&requestData)
+		suite.NoError(err)
+
+		// initialize the callback
+		err = keeper.DepositTxCallback(quicksilver.InterchainstakingKeeper, ctx, []byte("testing"), icqtypes.Query{ChainId: suite.chainB.ChainID, Request: resDataBz})
```

Comment:
I think we should separate cases nil and wrong bytes for testing. In case of wrong bytes, you just need to set `Request: []byte{}` in line 1737